src/main/jbake/content/blog/2017/03/17/apm-adjusting-sampling-rate.adoc

```diff
@@ -0,0 +1,385 @@
+= Adjusting sampling rates for Hawkular APM on OpenShift
+Juraci Paixão Kröhling
+2017-03-17
+:jbake-type: post
+:jbake-status: published
+:jbake-tags: blog, apm, opentracing, vertx, openshift, obsidian toaster
+
+When doing distributed tracing of a busy application, it's a good practice
+to limit the amount of data being collected or stored. The techniques for that
+are diverse, ranging from a simple "percentage based sampling" of the
+incoming requests, up to complex heuristics to keep/discard traces based on the
+code path.
+
+ifndef::env-github[]
+image::/img/blog/2017/2017-03-17-teaser.png[Final result]
+endif::[]
+ifdef::env-github[]
+image::../../../../../assets/img/blog/2017/2017-03-17-teaser.png[Final result]
+endif::[]
+
+It's also a good practice to empower the admin, or an automated monitoring system,
+with the capability of adjusting the sampling based on the current conditions.
+For instance, we'd probably want to get more traces from a
+link:/blog/2017/02/13/monitoring-canary-deployments.html[newly deployed version]
+of an existing application. We might want to compare the performance and adjust
+the percentage in steps. Similarly, if we are experiencing traffic peaks and need
```

Comment:
Not only for performance reasons: performance is one of the reasons people might be interested in implementing a sampling strategy.